src/server/server.zig

```diff
@@ -92,6 +93,30 @@ pub const User = struct { // MARK: User
 		}
 	}
 
+	pub fn damage(self: *User, dam: f32, cause: main.game.DamageType) void {
+		if (self.gamemode.load(.monotonic) == .creative) {
+			return;
+		}
+
+		self.player.health -= dam;
+
+		if (self.player.health <= 0) {
+			self.kill(cause);
+		}
+	}
+
+	pub fn kill(self: *User, cause: main.game.DamageType) void {
+		const message = switch (cause) {
+			.none => std.fmt.allocPrint(main.stackAllocator.allocator, "{s}§#ffffff died", .{self.name}) catch unreachable,
+			.fall => std.fmt.allocPrint(main.stackAllocator.allocator, "{s}§#ffffff died of fall damage", .{self.name}) catch unreachable,
```

Comment:
I would prefer to see the message in  the DamageType enum, then you could just do
```zig
const message = std.fmt.allocPrint(main.stackAllocator.allocator, "{s}§#ffffff {s}", .{self.name, self.cause.message()}) catch unreachable,
```